lib/plugins/aws/package/lib/mergeIamTemplates.js

```diff
@@ -30,10 +30,19 @@ module.exports = {
         },
       };
 
-      if ('logRetentionInDays' in this.serverless.service.provider) {
-        newLogGroup[logGroupLogicalId].Properties.RetentionInDays
-          = this.serverless.service.provider.logRetentionInDays;
+      if (_.has(this.serverless.service.provider, 'logRetentionInDays')) {
+        if (_.isInteger(this.serverless.service.provider.logRetentionInDays) &&
+        this.serverless.service.provider.logRetentionInDays > 0) {
+          if ('logRetentionInDays' in this.serverless.service.provider) {
```

Comment:
This (the old) if is now obsolete.

Reply:
@HyperBrain 
Oh :sweat_smile: 
How should I write?
I would like to ignore when provider don't have logRetentionInDays but throw error when wrong value is in

Reply:
The code is ok, just remove the most inner "if", because `_.has()` already checked for the existence.
So the inner if is useless - the code exactly does what you described above.

Reply:
I mean `if ('logRetentionInDays' in this.serverless.service.provider) {`

It already did the same check with `_.has(this.serverless.service.provider, 'logRetentionInDays')`

Reply:
Oh, it's accidentally. Thanks